client-mr/src/main/java/org/apache/hadoop/mapreduce/v2/app/RssMRAppMaster.java

```diff
@@ -201,21 +201,22 @@ public Thread newThread(Runnable r) {
       extraConf.set(RssMRConfig.RSS_REMOTE_STORAGE_CONF, remoteStorage.getConfString());
 
       // When containers have disk with very limited space, reduce is allowed to spill data to hdfs
-      if (extraConf.getBoolean(RssMRConfig.RSS_REDUCE_REMOTE_SPILL_ENABLED,
+      if (conf.getBoolean(RssMRConfig.RSS_REDUCE_REMOTE_SPILL_ENABLED,
         RssMRConfig.RSS_REDUCE_REMOTE_SPILL_ENABLED_DEFAULT)) {
 
-        if (storageType != StorageType.MEMORY_LOCALFILE_HDFS.name()) {
+        if (storageType != StorageType.MEMORY_LOCALFILE_HDFS.name()
+            || remoteStorage.isEmpty()) {
```

Comment:
```
if (remoteStorage.isEmpty())
```
we can look at the fetchRemoteStorage internal implement.

Reply:
Look at method `requireRemoteStorage`

Reply:
cleaned